### ISSUE TITLE
Simulation time (epochs) and transition probabilities workflow

### ISSUE THREAD

**Issue by kpelechrinis** [tnoulas/UrbanEpidemicSim#1]
If we discretize time we will have to consider some place features that will define the distribution of time a person stays at a given venue. So at each time slot the first decision will be "is user x transitioning?" (?) So this way we will have a set of people at each time slot in a location and we can focus on the people in the same location and see how infections propagate. 

The next question is that if we have a user that we decide they transition to another place, we have the place net, but this gives us only 5 top locations, we should have some randomness on going to one of the rest 85K places. Also the placenet from 2010-11 has all the transitions and the are timestamped (what are the timestamps? check-in times?)

**Comment by tnoulas** [comment]
Currently the setup is simple:
for every out transition of place i, we move a fraction of the population to place j. The fraction is 1/total_out_transitions_from_i . This is a very simple formulation. Also remember that we don't care about 'which' specific person moves, as all people are the same. That simplifies computations. Essentially we have population (set of people) exchanges between places at each epoch. 
See: https://github.com/tnoulas/UrbanEpidemicSim/blob/95013c4a0dd15d9ad964b1021c43be16a5e9c572/PlaceNetSim_v2.py#L109

Of course we can take into account variations of the above. Btw where is the top-5 locations coming from? 

timestamp 1 is check-in at place 1, and timestamp2 is check-in time at place 2. 

note also the code is not yet tested! My plan for tonight :)

**Comment by tnoulas** [comment]
When we add places artificially, eg. residences, we will adapt the population exchanges accordingly of course. This is just a very simple first model where the mass of people moving at each epoch is a fraction of the total observed movements. There are movements in reality we don't observe of course but i think the goal here would be start from the real data and then progressively add synthetic parts of make it more realistic. Does it make sense?

**Comment by kpelechrinis** [comment]
Yes makes sense. I mentioned the top-5 because I remember the next places from the API only returns the top-5 next venues and I thought that was the case with the dataset they released for the future cities (?) (but of course the 2010-11 data that you have are actual transitions so it is "complete" in that sense). I might be wrong though and I have to go back and check

**Comment by tnoulas** [comment]
No for this toy dataset, this is not true as we had collected all foursquare check-ins from twitter. That was around 20% of FSQ traffic at the time. After collecting user check-ins we simply added venue info from the api.

**Comment by kpelechrinis** [comment]
ok now I looked at the code and got a better understanding. For now the simulation is replaying exactly the moves of the dataset, right? There is not randomness in terms of the underlying movement, but only with terms to who is getting infected in each venue. I like starting with this and implementing the rest of the simulation engine first.

**Comment by tnoulas** [comment]
Yes this is the case exactly. Bare with me a few minutes, I have been solving a lot of bugs. I will push a new version on git in a bit and I will notify. 

Also happy with what you suggest about movement. Currently I am initialising the population of the place equal to the number of transitions that the place is giving, so the total population is around 1.2M. Of course these guys will move differently and will end up at residences etc. Where we put the residences is another question as we discussed ;)
